File: Scripts/newTask/factorial/perl/factorial.pl
foreach ( 0 .. 1000000 ) {
   factorial( 1000 );
}

sub factorial
{
   my $number = shift;   # get the argument

   if ( $number <= 1 ) { # base case
      return 1;
   }
   else {                # recursive step
      return $number * factorial( $number - 1 );
   }
}

File: Scripts/newTask/factorial/javascript/factorial.js
var num = 100;

//recursive
var factorial = function(n) {
    if(n == 0) {
        return 1
    } else {
        return n * factorial(n - 1);
    }
}
 
for (var i = 0; i < 1000000; ++i) { 
factorial(num);
}
